sm/sm-sbi.c

```diff
@@ -134,6 +134,74 @@ uintptr_t mcall_sm_call_plugin(uintptr_t plugin_id, uintptr_t call_id, uintptr_t
   return call_plugin(cpu_get_enclave_id(), plugin_id, call_id, arg0, arg1);
 }
 
+uintptr_t mcall_sm_send_msg(size_t uid, uintptr_t buf, size_t msg_size){
+  /* only an enclave itself can call this SBI */
+  enclave_ret_code ret;
+
+  if (!cpu_is_enclave_context()) {
+    return ENCLAVE_SBI_PROHIBITED;
+  }
+
+  ret = send_msg(cpu_get_enclave_id(), uid, (void *) buf, msg_size); 
+
+  return ret; 
+}
+
+uintptr_t mcall_sm_recv_msg(size_t uid, uintptr_t buf, size_t msg_size){
+  /* only an enclave itself can call this SBI */
+  enclave_ret_code ret;
+
+  if (!cpu_is_enclave_context()) {
+    return ENCLAVE_SBI_PROHIBITED;
+  }
+
+  ret = recv_msg(cpu_get_enclave_id(), uid, (void *) buf, msg_size); 
+
+  return ret; 
+}
+
+
+uintptr_t mcall_sm_mem_share(size_t uid, uintptr_t enclave_addr, uintptr_t enclave_size){
```

Comment:
Can you change the `mem_share` and `mem_stop` to `mem_share_start` and `mem_share_stop`?